testdata/p4_16_samples_outputs/pna-dpdk-header-stack-assignment.p4.spec

```diff
@@ -74,9 +74,48 @@ action NoAction args none {
 }
 
 action encap_one_tunnel_layer_ipv4 args instanceof encap_one_tunnel_layer_ipv4_arg_t {
-	mov h.ipv4_3 h.ipv4_2
-	mov h.ipv4_2 h.ipv4_1
-	mov h.ipv4_1 h.ipv4_0
+	jmpnv LABEL_FALSE_0 h.ipv4_2
+	validate h.ipv4_3
```

Comment:
It appears that the pseudocode for what you are doing in the expansion is like this:
```
if (h.ipv4_2.isValid()) {
    h.ipv4_3.setValid();
} else {
    h.ipv4_3.setInvalid();
}
h.ipv4_3.dscp_ecn = h.ipv4_2.dscp_en;
h.ipv4_3.length = h.ipv4_2.length;
// etc. copy other header fields here
```
but this would be more efficient in the case when the header being copied from is invalid, yes?
```
if (h.ipv4_2.isValid()) {
    h.ipv4_3.setValid();
    h.ipv4_3.dscp_ecn = h.ipv4_2.dscp_en;
    h.ipv4_3.length = h.ipv4_2.length;
    // etc. copy other header fields here
} else {
    h.ipv4_3.setInvalid();
}
```

Reply:
Note: If you would prefer to approve and merge in this bug fix first, and create a separate issue to consider optimizing later, let me know.  I will go ahead and approve this PR.

Reply:
I have created this issue to track the possible future optimization that can be done here for invalid headers: https://github.com/p4lang/p4c/issues/4155 so I can merge in this fix now.